packages/react-ai/src/hooks/useAIConversation.tsx

```diff
@@ -0,0 +1,188 @@
+import React from 'react';
+import { V6Client } from '@aws-amplify/api-graphql';
+import {
+  Conversation,
+  ConversationMessage,
+} from '@aws-amplify/data-schema/dist/esm/ai/ConversationType';
+import { DataState } from '@aws-amplify/ui-react-core';
+import { useAIContext } from './AIContextProvider';
+
+interface SendMesageParameters {
+  content: Parameters<Conversation['sendMessage']>[0]['content'];
+  aiContext?: Parameters<Conversation['sendMessage']>[0]['aiContext'];
+  responseComponents?: any;
+}
+
+interface UseAIConversationInput {
+  id?: string; // should attempt to create a new session id if none is passed
+  onResponse?: (message: ConversationMessage) => void;
+}
+
+interface AIConversationState {
+  messages: ConversationMessage[];
+}
+
+export type UseAIConversationHook<T extends string> = (
+  routeName: T,
+  input?: UseAIConversationInput
+) => [DataState<AIConversationState>, (input: SendMesageParameters) => void];
+
+export function createUseAIConversation<T extends V6Client<any>>(
+  client: T
+): UseAIConversationHook<Extract<keyof T['conversations'], string>> {
+  const useAIConversation: UseAIConversationHook<
+    Extract<keyof T['conversations'], string>
+  > = (routeName: keyof T['conversations'], input = {}) => {
+    const clientRoute = (client.conversations as T['conversations'])[routeName];
+
+    const { routeToConversationsMap, setRouteToConversationsMap } =
+      useAIContext();
+    const messagesFromAIContext = input.id
+      ? routeToConversationsMap[routeName as string]?.[input.id]
+      : undefined;
+    const [localMessages, setLocalMessages] = React.useState<
+      ConversationMessage[]
+    >(messagesFromAIContext ?? []);
```

Comment:
If we are using an AIContextProvider, the hooks local message state might be updated from the provider (another hook sending a message), this helps capture that